lib/sensu-plugins-graphite/graphite_proxy/proxy.rb

```diff
@@ -79,7 +79,7 @@ def retrieve_data!
               config[:server].prepend('http://')
             end
 
-            url = "#{config[:server]}/render?format=json&target=#{formatted_target}&from=#{config[:from]}"
+            url = "#{config[:server]}/render?format=json&target=#{formatted_target}&from=#{config[:from]}&until=#{config[:until]}"
```

Comment:
From what I read [here](https://graphite.readthedocs.io/en/latest/render_api.html) it does not appear to be a quired field and I am pretty sure if someone does not specify that it will evaluate to nil or brick. I think we should probably do something like this:
```ruby
url = if !config[:until].nil?
        "#{config[:server]}/render?format=json&target=#{formatted_target}&from=#{config[:from]}&until=#{config[:until]}"
      else
        "#{config[:server]}/render?format=json&target=#{formatted_target}&from=#{config[:from]}"
      end
```

That being said I don't know that `from` is either we might want to create a helper function that loops through the list of defined variables and adds them to the string. Something along these lines:
```ruby
def build_render_query(config)
  url = "#{config[:server]}/render?format=json&target=#{formatted_target}"
  if !config[:from].nil?
    url += "&from=#{config[:from]}"
  if !config[:until].nil?
    url += "&until=#{config[:until]}"
  end
  url
end
```

then you can set `url = build_render_query(config)` where the current url definition is.

@eheydrick thoughts?

Reply:
@majormoses FWIW: if you specify either the ``from`` or ``until`` parameters without a value, it still works as expected. See https://github.com/graphite-project/graphite-web/blob/2a897cf15a1ad9beb2f919267ef98ce40271bea6/webapp/graphite/metrics/views.py#L75:L83 for the code. Also tested on a live Graphite instance.

Actually a proper fix would be:

* filter out nil values (although it doesn't appear to be necessary).
* build the actual query string component using [URI.encode_www_form()](https://docs.ruby-lang.org/en/2.4.0/URI.html#method-c-encode_www_form) so everything is properly encoded.

@inno-ron Can I help you with this PR?